### DISCUSSION TITLE
Proxy for websocket without the context does not work.

### DISCUSSION THREAD

**Discussion by garnesen** [chimurai/http-proxy-middleware#582] [Q&A]
Proxying works fine when I use
```javascript
const app = express();
const proxy = require('http-proxy-middleware');
const wsProxy = proxy('/socket.io', {target:'ws://my-websocket.com/', changeOrigin: true, ws:true});
app.use(wsProxy);
```
but it does not work if I use
```javascript
const app = express();
const proxy = require('http-proxy-middleware');
app.use('/socket.io', proxy({target:'ws://my-websocket/', changeOrigin: true, ws:true}));
```

Am I doing this wrong, or is this a bug?

**Comment by garnesen** [comment]
Looks similar to this issue https://github.com/chimurai/http-proxy-middleware/issues/204

**Comment by garnesen** [comment]
After doing some research, it seems with the not working example, express never hits the middleware function at https://github.com/chimurai/http-proxy-middleware/blob/bb59c666dbb65cc0ef422bbf5332b67cbd6fb0e3/lib/index.js#L38

This means `catchUpgradeRequest` is never called and therefore no handler is set for upgrade and the proxy is never made. In the first working example, the middleware is called for all requests so the upgrade handler is set.

Why express middleware not working for /socket.io? I do not know..

**Comment by garnesen** [comment]
I tried using plain [Websockets](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) in case it was a socket.io problem and it looks like express middleware does not seem to catch any of it. I saw there are libraries to add middleware for websockets but it seems like a solution to my problem is to not use middleware at all. I can simply do

```javascript
const app = express();
const proxy = require('http-proxy-middleware');
const wsProxy = proxy('/socket.io', {target:'ws://my-websocket.com/', changeOrigin: true, ws:true});
const server = app.listen(...);
server.on('upgrade', wsProxy.upgrade);
```

**Comment by sincerefly** [comment]
@garnesen 

Add `app.use(wsProxy);` and it's works well, thanks

**Reply by maitrungduc1410** [reply]
thank you, it's weird :D
